### DISCUSSION TITLE
3D panel creation locally

### DISCUSSION THREAD

**Discussion by pppaw** [foxglove/community#811] [General]
Hello,

I have downloded 3D panel src files from foxglove github open source to built new 3D panel locally and following the extension panel steps to built it locally. I am getting error at the time of debugging  as:

Cannot find module '@foxglove/studio-base/components/Stack' or its corresponding type declarations.

How to solve this error?

**Comment by jtbandes** [comment]
The 3D panel (and other built-in panels) uses some internal shared components and weren't written to be able to build totally independently of the studio repo. If you want to build a .foxe extension, the extension needs to be bundled independently and can't access internal components from the app at runtime.
